File: tps/tp1/PalindromoRec.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool palindromo(char palavra[], int i, int j) {
    if (i >= j) {
        return true;
    } else if (palavra[i] != palavra[j]) {
        return false;
    } else {
        return palindromo(palavra, i + 1, j - 1);
    }
}
int main() {
    char palavra[1000];

    while (true) {
        scanf("%s", palavra);

        if (strcmp(palavra, "FIM") == 0) {
            break;
        } else {
            int len = strlen(palavra);
            if (palindromo(palavra, 0, len - 1)) {
                printf("SIM\n");
            } else {
                printf("NAO\n");
            }
        }
    }
    return 0;
}

File: tps/tp1/Questao 10/PalindromoRec.java
package tps.tp1;

import java.util.Scanner;

public class PalindromoRec {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String string;

        while (!(string = sc.nextLine()).equals("FIM")) {
            if (palindromo(string, 0, string.length() - 1)) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }
        }
        sc.close();
    }

    public static boolean palindromo(String string, int i, int j) {
        if (i >= j) {
            return true;
        } else if (string.charAt(i) != string.charAt(j)) {
            return false;
        } else {
            return palindromo(string, i + 1, j - 1);
        }
    }
}
